membership-attribute-service/app/controllers/AccountController.scala

```diff
@@ -300,14 +300,30 @@ class AccountController(commonActions: CommonActions, override val controllerCom
       result <- if (records.isEmpty)
         OptionEither.liftFutureEither[Subscription[AnyPlan]](None)
       else
-        OptionEither.liftFutureOption(subscriptionService.get[AnyPlan](Name(records.head.Name), isActiveToday = true))
+        reuseAlreadyFetchedSubscriptionIfAvailable(records, nonGiftSubs, subscriptionService)
     } yield result
     val fullList = giftSub
-      .map(sub => ContactAndSubscription(contact, sub) :: nonGiftSubs)
+      .map(sub => ContactAndSubscription(contact, sub, isGiftRedemption = true) :: nonGiftSubs)
       .getOrElse(nonGiftSubs)
     OptionEither.liftOption(fullList.run)
   }
 
+  def reuseAlreadyFetchedSubscriptionIfAvailable(
+    giftRecords:  List[ZuoraRestService.GiftSubscriptionsFromIdentityIdRecord],
+    nonGiftSubs: List[ContactAndSubscription],
+    subscriptionService: SubscriptionService[Future],
+  ) = {
+    val subscriptionName = Name(giftRecords.head.Name)
+    OptionEither.liftFutureOption(
+      // If the current user is both the gifter and the giftee we will have already retrieved their
+      // subscription so we can reuse it and avoid a call to Zuora
```

Comment:
is this an important optimisation considering the likeihood of it happening, multiplied by the chance of the person going into MMA regularly enough to slow things down?

Reply:
If it wasn't for the fact that we are constantly over quota on our requests to Zuora during the day I'd say no, but given that we are I think it's worth reducing the number of calls we make wherever possible.